modules/product/src/main/java/com/opengamma/strata/product/swap/type/InflationRateSwapLegConvention.java

```diff
@@ -121,15 +121,17 @@
    * 
    * @param index  the index, the market convention values are extracted from the index
    * @param lag  the lag between the price index and the accrual date, typically a number of months
+   * @param priceIndexCalculationMethod the price index calculation method, typically interpolated or monthly
    * @param businessDayAdjustment the business day 
    * @return the convention
    */
   public static InflationRateSwapLegConvention of(
       PriceIndex index,
       Period lag,
+      PriceIndexCalculationMethod priceIndexCalculationMethod,
```

Comment:
Ok. Have deprecated the old method as the client should have to provide interpolation style when defining conventions. Defaulting to monthly seems slightly arbitrary.